src/skai/beam_utils.py

```diff
@@ -107,6 +108,8 @@ def get_pipeline_options(
     max_workers: int,
     worker_service_account: str | None,
     worker_type: str | None,
+    worker_machine_type: str | None,
+    dataflow_service_options: str
```

Comment:
@panford Are there specific options we need to add here to enable GPU/TPU support? Can we use a narrower parameter here (e.g. "accelerator_type") but set "dataflow_service_options" correctly?

Reply:
Can you also add the new parameters to the docstring for this function? +1 that "worker_machine_type" could be renamed to be more narrow since there is already flag "worker_type"

Reply:
Hi @jzxu 

This is how I run the inference code from terminal now. The worker accelerator gets set here together with the gpu count in the dataflow service options flag.

```
#!/bin/bash

export GOOGLE_APPLICATION_CREDENTIALS=../***.json
export EXAMPLES_PATTERN=gs://skai-project/hurricane_ian/skai_dataset/skai_dataset/1.0.0/skai_dataset-labeled_test.tfrecord-*
export OUTPUT_PATH=gs://skai-project/experiments/inference_output
export MODEL_DIR=gs://skai-project/experiments/model
export PYTHONPATH=./src

WORKER_ACCELERATOR='nvidia-tesla-t4'
GPU_COUNT=1
DATAFLOW_SERVICE_OPTIONS="$WORKER_ACCELERATOR;count:$GPU_COUNT;install-nvidia-driver"

python src/skai/model/inference.py \
  --examples_pattern=$EXAMPLES_PATTERN \
  --output_path=$OUTPUT_PATH \
  --model_dir=$MODEL_DIR \
  --use_dataflow \
  --cloud_project=skai-proj \
  --cloud_region=us-central1 \
  --dataflow_temp_dir='gs://skai-project/disaster-assessment/dataflow-temp' \
  --worker_service_account='skai-dataflow@***.iam.gserviceaccount.com' \
  --max_dataflow_workers=4 \
  --batch_size=64 \
  --machine_type='n1-highmem-8' \
  --dataflow_service_options=$DATAFLOW_SERVICE_OPTIONS
```

Reply:
I guess we set the dataflow service options depending on the chosen accelerator?

@jlee24 Will go ahead to resolve the conflict between `worker_machine_type` and `worker_type` flags and docstring also.

Reply:
@panford Thanks for clarifying the dataflow_service_options settings! I propose that we let the user select the accelerator they want to use (if any) and the count, and create the dataflow_service_options string in the code, as follows:

```
def get_pipeline_options(
    use_dataflow: bool,
    job_name: str,
    project: str,
    region: str,
    temp_dir: str,
    max_workers: int,
    worker_service_account: str | None,
    machine_type: str | None,
    accelerator: str | None,
    accelerator_count: int
) -> PipelineOptions:
  """Returns dataflow pipeline options.

  Args:
    use_dataflow: Whether to use Dataflow or local runner.
    job_name: Name of Dataflow job.
    project: GCP project.
    region: GCP region.
    temp_dir: Temporary data location.
    max_workers: Maximum number of Dataflow workers.
    worker_service_account: Email of the service account will launch workers.
        If None, uses the project's default Compute Engine service account
        (<project-number>-compute@developer.gserviceaccount.com).
    machine_type: Dataflow worker type.
    accelerator: Accelerator type, e.g. nvidia-tesla-t4.
    accelerator_count: Number of acccelerators to use.

  Returns:
    Dataflow options.
  """
  if not use_dataflow:
    return PipelineOptions.from_dictionary({
        'runner': 'DirectRunner',
        'direct_num_workers': 10,
        'direct_running_mode': 'multi_processing',
    })

  if not project or not region:
    raise ValueError(
        'cloud_project and cloud_region must be specified when using '
        'Dataflow.')

  options = {
      'job_name': job_name,
      'project': project,
      'region': region,
      'temp_location': temp_dir,
      'runner': 'DataflowRunner',
      'experiments': 'use_runner_v2',
      'sdk_container_image': _get_dataflow_container_image(),
      'sdk_location': 'container',
      'setup_file': _get_setup_file_path(),
      'max_num_workers': max_workers,
      'use_public_ips': False,  # Avoids hitting public ip quota bottleneck.
  }
  if worker_service_account:
    options['service_account_email'] = worker_service_account
  if worker_type:
    options['machine_type'] = worker_type

  if accelerator:
    dataflow_service_options = f'{accelerator};count:{accelerator_count}'
    if 'nvidia' in accelerator:
      dataflow_service_options += ';install-nvidia-driver'
    options['dataflow_service_options'] = dataflow_service_options

  return PipelineOptions.from_dictionary(options)
```

WDYT?

Reply:
@jzxu That should do it